graphprot/NeuralNet.py

```diff
@@ -99,23 +139,47 @@ def __init__(self, database=None, Net=None,
             self.valid_loader = DataLoader(
                 valid_dataset, batch_size=self.batch_size, shuffle=self.shuffle)
             print('Independent validation set loaded')
-            if cluster_nodes == True :
-                PreCluster(valid_dataset, method='mcl')
-            
+            if self.cluster_nodes == 'mcl' or self.cluster_nodes == 'louvain':
+                PreCluster(valid_dataset, method=self.cluster_nodes)
+
         else:
             print('No independent validation set loaded')
 
+        self.put_the_model(dataset, Net)
+
+        # optimizer
+        self.optimizer = torch.optim.Adam(
+            self.model.parameters(), lr=self.lr)
+
+        self.set_loss()
+
+        # init lists
+        self.train_acc = []
+        self.train_loss = []
+
+        self.valid_acc = []
+        self.valid_loss = []
+
+    def put_the_model(self, dataset, Net):
```

Comment:
small thing but maybe the name of this function should be b it more clear (put_model_to_device or something)